lib/portlayer/storage/vsphere/datastore.go

```diff
@@ -301,3 +303,26 @@ func (d *Datastore) mkRootDir(ctx context.Context, rootdir string) error {
 func (d *Datastore) rootDir() string {
 	return strings.SplitN(d.RootURL, " ", 2)[1]
 }
+
+// Parse the datastore format ([datastore1] /path/to/thing) to groups.
+var datastoreFormat = regexp.MustCompile(`^\[([\w\d\(\)\s]+)\]`)
+var pathFormat = regexp.MustCompile(`\s[\/\w]+$`)
+
+// Converts `[datastore] /path` to URL
+func DatastoreToURL(ds string) (*url.URL, error) {
+	u := new(url.URL)
+	u.Host = strings.TrimSuffix(strings.TrimPrefix(datastoreFormat.FindString(ds), "["), "]")
```

Comment:
Reading further, I think you need the `Trim` because you're using `FindString`.  Why not use one of the `Match` calls which returns the result of the match.